### ISSUE TITLE
Improve 'trusted-set-cookie' / 'trusted-set-local-storage-item' — add ability to specify base64 as encoding

### ISSUE THREAD

**Issue by contribucious** [AdguardTeam/Scriptlets#307]
### _Steps to reproduce_

1. Add this user rule:
```adblock
@@||www.coolblue.be^$important,document
```
2. Go to https://www.coolblue.be/en and [refuse cookies](https://user-images.githubusercontent.com/4764956/230755586-3b52a7e8-4529-402c-b671-b8ce3df5a483.png).
3. Observe the apparition of a new _**cookie-preferences**_ cookie with this value:
```
eyJmdW5jdGlvbmFsIjp0cnVlLCJhbmFseXRpY2FsIjp0cnVlLCJtYXJrZXRpbmciOmZhbHNlfQ%3D%3D
```

### _Goal_

Replacing this rule:
```adblock
! Base64 value for: {"functional":true,"analytical":true,"marketing":false}
coolblue.be#%#//scriptlet('trusted-set-cookie', 'cookie-preferences',
'eyJmdW5jdGlvbmFsIjp0cnVlLCJhbmFseXRpY2FsIjp0cnVlLCJtYXJrZXRpbmciOmZhbHNlfQ==')
```

By this one:
```adblock
coolblue.be#%#//scriptlet('trusted-set-cookie', 'cookie-preferences',
'base64:{"functional":true,"analytical":true,"marketing":false}')
```
&nbsp;
### _:writing_hand: Remarks_

1. Possibly use a dedicated parameter instead.
2. Consider a variant that does not use [`encodeURIComponent`](https://github.com/AdguardTeam/Scriptlets/blob/8edd4ec246a3acac608d67975610de11f9b0d960/src/helpers/cookie-utils.js#L37) for storing **base64** value,
for some websites reading the string _directly_ in `atob`.
    - That is, the stored value will therefore include `==` instead of `%3D%3D` at the end.
3. Should ideally also apply to `trusted-set-local-storage-item`.

&nbsp;
### Hi and thanks in advance! :v:

**Comment by contribucious** [comment]
> 1\. Possibly use a dedicated parameter instead.

💡 Or … a new "comma-separated" `advancedOptions` at the end ([code](https://github.com/AdguardTeam/Scriptlets/blob/402972c3ca6c11e075aa576ee88bec34d9752e24/src/scriptlets/trusted-set-cookie.js#L71) and [wiki](https://github.com/AdguardTeam/Scriptlets/blob/master/wiki/about-trusted-scriptlets.md#trusted-set-cookie)), which can include `as={plaintext|base64}` or maybe `encoding={none|base64}` — the former in the sense of _"to store value as …"_, of course.

↪️ Probably both more flexible (especially to add support for the described variant in addition) and more future-proof. ☺️

**Comment by Alex-302** [comment]
But we can convert a scring to BASE64 using tools.

**Comment by contribucious** [comment]
Of course. Even web tools, easily. But the goal is to have **_self-explanatory_ rules** for these cases (that is, no need to have an _additional_ comment describing the cryptic content to readers).

↬ It's mainly for adding clarity and transparency to **public filters** (i.e. those read by people), more so than for local filters. 🙂

**Comment by contribucious** [comment]
Issue clarified just now. ✔️
